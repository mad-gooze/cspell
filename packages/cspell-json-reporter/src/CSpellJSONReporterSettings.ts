/**
 * CSpell-json-reporter settings type definition
 */
export type CSpellJSONReporterSettings = {
    /**
     * Output JSON file path
     */
    outFile: string;
    /**
     * Add more information about the files being checked and the configuration
     * @default false
     */
    verbose?: boolean;
    /**
     * Add information useful for debugging cspell.json files
     * @default false
     */
    debug?: boolean;
    /**
     * Add progress messages
     * @default false
     */
    progress?: boolean;
};
