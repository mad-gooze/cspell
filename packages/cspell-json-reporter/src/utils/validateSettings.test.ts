import { CSpellJSONReporterSettings } from "../CSpellJSONReporterSettings";
import { validateSettings } from "./validateSettings";

describe('validateSettings', () => {
    it('passes valid settings', () => {
        const valid: CSpellJSONReporterSettings[] = [
            { outFile: 'foobar' },
            { outFile: 'foobar', verbose: true },
            { outFile: 'foobar', debug: false },
            { outFile: 'foobar', progress: true },
            { outFile: 'foobar', verbose: undefined },
        ];
        valid.forEach((settings) => {
            expect(validateSettings(settings)).toEqual(undefined);
        });
    });

    it('throws for invalid settings', () => {
        const invalid = [
            { },
            { outFile: 1 },
            { outFile: 'foobar', verbose: 123 },
            { outFile: 'foobar', debug: [] },
        ];
        invalid.forEach((settings) => {
            expect(() => validateSettings(settings)).toThrowErrorMatchingSnapshot();
        });
    });
});
