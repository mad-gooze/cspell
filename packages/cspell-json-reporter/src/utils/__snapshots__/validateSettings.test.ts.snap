// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateSettings throws for invalid settings 1`] = `"cspell-json-reporter settings.outFile must be a string"`;

exports[`validateSettings throws for invalid settings 2`] = `"cspell-json-reporter settings must be an object"`;

exports[`validateSettings throws for invalid settings 3`] = `"cspell-json-reporter settings.outFile must be a string"`;

exports[`validateSettings throws for invalid settings 4`] = `"cspell-json-reporter settings.outFile must be a string"`;

exports[`validateSettings throws for invalid settings 5`] = `"cspell-json-reporter settings.verbose must be a boolean"`;

exports[`validateSettings throws for invalid settings 6`] = `"cspell-json-reporter settings.debug must be a boolean"`;
